### PR TITLE
OpenApi Audit Messages manager implemented

### PR DESCRIPTION
**Pull request by xpavlic**
- service manager methods implemented
- added new schemas, parameter and response

### CODE REVIEW

File: perun-openapi/openapi.yml [hunk 1]
```diff
@@ -1771,6 +1802,10 @@ tags:
     description: AuthzResolver RPC API in Perun
     externalDocs:
       url: https://perun-aai.org/documentation/technical-documentation/rpc-api/rpc-javadoc-AuthzResolver.html
+  - name: AuditMessagesManagerMethod
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]:
```suggestion
  - name: AuditMessagesManager
```

File: perun-openapi/openapi.yml [hunk 2]
```diff
@@ -202,6 +202,21 @@ components:
...
+        message: { type: string }
+
+    AuditMessage:
+      allOf:
+        - $ref: '#/components/schemas/AuditEvent'
+        - properties:
+            id: { type: integer }
+            actor: { type: string }
+            createdAt: { type: string }
+            createdByUid: { type: integer }
```
[the remark this answers is not in the document]
Comment (Vojtech-Sassmann) [reply perun-openapi/openapi.yml]: This is how is written composition in OpenAPI. The way you had it is how is done inheritance.

File: perun-openapi/openapi.yml [hunk 3]
```diff
@@ -4859,6 +4894,179 @@ paths:
...
+      operationId: createAuditerConsumer
+      summary: |
+        Creates new auditer consumer with last processed id which equals current auditer log max id.
+      parameters:
+        - name: consumerName
+          in: query
+          schema:
+            type: string
+          description: New name for consumer
+          required: true
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]:
```suggestion
        - $ref: '#/components/parameters/consumerName'
```

File: perun-openapi/openapi.yml [hunk 4]
```diff
@@ -4859,6 +4894,179 @@ paths:
...
+        Returns 100 newest audit messages from audit log. If there is a less messages than 100,
+        then all of them are returned OR Returns exact number of newest audit messages defined by
+        'count' param (disregarding message IDs). If there is less messages present, then all of them are returned..
+      parameters:
+        - name: count
+          in: query
+          schema:
+            type: integer
+          description: Messages limit
+          required: false
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]: Please use the short syntax to save lines:
```suggestion
        - { name: count, in: query, schema: { type: integer },  description: "Messages limit",  required: false }
```

File: perun-openapi/openapi.yml [hunk 5]
```diff
@@ -4859,6 +4894,179 @@ paths:
...
+        Returns all messages with IDs within the range from max(ID) to (max(ID)-count),
+        where number of returned messages is equal or less than 'count' param, because some
+        IDs could be skipped in the sequence.
+      parameters:
+        - name: count
+          in: query
+          schema:
+            type: integer
+          description: Number of IDs to subtract from max_id
+          required: true
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]:
```suggestion
        - { name: count, in: query, schema: { type: integer }, description: "Number of IDs to subtract from max_id", required: true }
```

File: perun-openapi/openapi.yml [hunk 6]
```diff
@@ -4859,6 +4894,179 @@ paths:
...
+      summary: |
+        Set ID of last processed message for specified consumer.
+      parameters:
+        - $ref: '#/components/parameters/consumerName'
+        - name: lastProcessedId
+          in: query
+          schema:
+            type: integer
+          description: id of message to what consumer will be set
+          required: true
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]:
```suggestion
        - { name: lastProcessedId, in: query, schema: { type: integer }, description: "id of message to what consumer will be set",  required: true }
```

File: perun-openapi/openapi.yml [hunk 7]
```diff
@@ -4859,6 +4894,179 @@ paths:
...
+      operationId: auditMessagesManagerLog
+      summary: |
+        Log arbitrary auditer message/event to the audit log.
+      parameters:
+        - name: msg
+          in: query
+          schema:
+            type: string
+          description: Message to be logged
+          required: true
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]:
```suggestion
        -  { name: msg, in: query, schema: { type: string }, description: "Message to be logged", required: true }
```

**Pull request merged**